byterun/caml/hooks.h

```diff
@@ -17,22 +17,92 @@
 #define CAML_HOOKS_H
 
 #include "misc.h"
-#include "memory.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 #ifdef CAML_INTERNALS
 
-#ifdef NATIVE_CODE
+/* Some macros to help declaring, exporting and using hooks */
+#define EXTERN_HOOK1(hook_name, arg1)      \
+  CAMLextern void (*hook_name)(arg1);
+#define DECLARE_HOOK1(hook_name, arg1)     \
+  CAMLexport void (*hook_name)(arg1) = NULL;
+#define MAYBE_HOOK1(hook_name, arg1) \
+  { if( hook_name != NULL ){ hook_name(arg1); } }
```

Comment:
`hook_name` should probably be guarded by brackets here, as it is used within a macro.